ShellPkg/Application/Shell/ShellProtocol.c

```diff
@@ -2426,14 +2469,19 @@ ShellSearchHandle (
       //
       // We want the root node.  create the node.
       //
-      FileInfo     = FileHandleGetInfo (FileHandle);
-      NewShellNode = CreateAndPopulateShellFileInfo (
-                       MapName,
-                       EFI_SUCCESS,
-                       L"\\",
-                       FileHandle,
-                       FileInfo
-                       );
+      FileInfo = FileHandleGetInfo (FileHandle);
+      // MU_CHANGE [START] - CodeQL change
+      if (FileInfo != NULL) {
+        NewShellNode = CreateAndPopulateShellFileInfo (
+                         MapName,
+                         EFI_SUCCESS,
+                         L"\\",
+                         FileHandle,
+                         FileInfo
+                         );
+      }
+
+      // MU_CHANGE [END] - CodeQL change
       SHELL_FREE_NON_NULL (FileInfo);

```

Comment:
Maybe this macro handles it, but should the free be inside the if FileInfo != NULL?